src/main/java/org/klodnicki/service/BookService.java

```diff
@@ -85,4 +91,32 @@ public List<String> prepareListOfBooks(String title, String author) {
     private List<BookInfo> getBooksInfoFromDatabaseByTitleAndAuthor(String title, String author) {
         return findBooksByTitleAndAuthor(title, author);
     }
+
+    private List<BookInfo> getAllBooksFromDatabaseSortByParameter(String parameter) {
+        return bookRepository.findAllBooksSortByParameter(parameter);
+    }
+
+
+    public List<String> prepareListOfAllBooksSortByParameter(String parameter) throws NotFoundInDatabaseException,
```

Comment:
Making this method throws IllegalArgumentException makes this code unreadable for me. What exactly throws this? This is runtime (unchecked) exception which you later try to catch. We talked about it and you rather should not do that, especially in such complicated method! I don't know what piece of code exactly is causing the throw of this exception.

Reply:
Ok I understand it. Method corrected.